torch/utils/data/datapipes/iter/combinatorics.py

```diff
@@ -145,6 +144,8 @@ def reset(self) -> None:
         self._buffer = []
         if self._enabled and self._seed is None:
             self._seed = int(torch.empty((), dtype=torch.int64).random_().item())
+        self._rng.seed(self._seed)
+        self._seed = None
```

Comment:
Thanks for the great PR! 

I'm wondering why we need to set self._seed to None here. I encountered a problem where iterating twice caused the dataset to lose determinism.

```python
dp1 = IterableWrapper(range(10)).shuffle().set_seed(1) 
dp2 = IterableWrapper(range(10)).shuffle().set_seed(1)

# First iteration
all_match1 = all(x == y for x, y in zip(dp1, dp2))  # true
# Second iteration
all_match2 = all(x == y for x, y in zip(dp1, dp2))  # false
```
I would appreciate any clarification you can provide!